hw/bsp/esp32s2_kaluga_1/esp32s2_kaluga_1.c

```diff
@@ -64,18 +94,22 @@ void board_init(void)
   gpio_set_direction(BUTTON_PIN, GPIO_MODE_INPUT);
   gpio_set_pull_mode(BUTTON_PIN, BUTTON_STATE_ACTIVE ? GPIO_PULLDOWN_ONLY : GPIO_PULLUP_ONLY);
 
-  // USB Controller Hal init
-  periph_module_reset(PERIPH_USB_MODULE);
-  periph_module_enable(PERIPH_USB_MODULE);
+  if (USB_WRAP.date.val == USBDC_PERSIST_ENA) {
+    // Enable USB/IO_MUX peripheral reset, if coming from persistent reboot
+    REG_CLR_BIT(RTC_CNTL_USB_CONF_REG, RTC_CNTL_IO_MUX_RESET_DISABLE);
+    REG_CLR_BIT(RTC_CNTL_USB_CONF_REG, RTC_CNTL_USB_RESET_DISABLE);
+  } else {
+    // Reset USB module
+    periph_module_reset(PERIPH_USB_MODULE);
+    periph_module_enable(PERIPH_USB_MODULE);
+  }
```

Comment:
@hathach I have discussed and agreed with our IDF team to move this code DCD. Do you have anything against?

Reply:
this is OK to move into DCD, as long as the dcd doesn't depend on the BSP (e.g pin/port mapping etc ..).